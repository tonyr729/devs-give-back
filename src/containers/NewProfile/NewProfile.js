import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

import './NewProfile.css';

export class NewProfile extends Component {
  
  displayCategories = (categories) => {
    let display = null;
    if (categories) {
      display = categories.map((category, index) => {
        return (
          <button key = {index} className="display-category">{category}</button>
        );
      });
    }
    return display;
  }
  
  logInCheck = (client) => {
    const value = Object.keys(client).length;
    if ( value === 0 ) {
      return <Redirect to='/client-login'/>;
    }
  }
  
  render() {
    const redirect = this.logInCheck(this.props.client);
    const categories = this.displayCategories(this.props.problem.categories);
    return (
      <div className="frame-container">
        {redirect}
        <div className="header">
          <div className="header-container">
            <p>{this.props.client.name}</p>
            <img className='profile-picture' src={this.props.client.photoURL} alt="user profile picture"/>
          </div>
        </div>
        <div className="problem">
          <div className="problem-container">
            <p className="section-title">Heres your problem</p>
            <p className="problem-title">{this.props.problem.title}</p>
            <p className="problem-body">{this.props.problem.body}</p>
            <div className="category-tags">
              {categories}
            </div>
          </div>
        </div>
        <div className="stats">
          <div className="stats-container">
            <p className="section-title">Heres whats happening</p>
            <p className="stats-title">Nothing at this time :(</p>
          </div>
        </div>
      </div>
      
    );
  }
}

export const mapStateToProps = (state) => ({
  client: state.client,
  problem: state.problem
});

NewProfile.propTypes = {
  client: PropTypes.object,
  problem: PropTypes.object
};


export default connect(mapStateToProps)(NewProfile);