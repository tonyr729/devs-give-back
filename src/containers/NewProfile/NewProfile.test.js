import { NewProfile, mapStateToProps, mapDispatchToProps} from './NewProfile';
import { shallow } from 'enzyme';
import React from 'react';
import { Redirect } from 'react-router-dom';


describe('NewProfile', () => {
  it('should match snapshot', () => {
    const mockProps = {
      client: {
        name: 'Tony'
      },
      problem: {
        categories: ['category']
      }
    }
    const wrapper = shallow(<NewProfile {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('logInCheck', () => {
    let wrapper;
    let mockProps

    beforeEach(() => {
      mockProps = {
        client: {},
        problem: {
          categories: ['category']
        }
      };
      wrapper = shallow(<NewProfile {...mockProps} />)
    });
    
    it('should redirect if there is no client', () => {
      
      const actual = wrapper.instance().logInCheck(mockProps.client)
      expect(actual).toEqual(<Redirect to='/client-login'/>)
    });
  });

  describe('mapStateToProps', () => {
    it('returns an object with client info', () => {

      const mockState = {
        client: {name: 'Tony'},
        problem: {
          categories: ['category']
        }
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
      
    });
  });
});