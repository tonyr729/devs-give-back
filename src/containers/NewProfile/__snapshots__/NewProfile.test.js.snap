// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProfile should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewProfile
    client={
        Object {
            "name": "Tony",
          }
    }
    problem={
        Object {
            "categories": Array [
              "category",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="header"
>
          <div
                    className="header-container"
          >
                    <p>
                              Tony
                    </p>
                    <img
                              alt="user profile picture"
                              className="profile-picture"
                              src={undefined}
                    />
          </div>
</div>,
        <div
          className="problem"
>
          <div
                    className="problem-container"
          >
                    <p
                              className="section-title"
                    >
                              Heres your problem
                    </p>
                    <p
                              className="problem-title"
                    />
                    <p
                              className="problem-body"
                    />
                    <div
                              className="category-tags"
                    >
                              <button
                                        className="display-category"
                              >
                                        category
                              </button>
                    </div>
          </div>
</div>,
        <div
          className="stats"
>
          <div
                    className="stats-container"
          >
                    <p
                              className="section-title"
                    >
                              Heres whats happening
                    </p>
                    <p
                              className="stats-title"
                    >
                              Nothing at this time :(
                    </p>
          </div>
</div>,
      ],
      "className": "frame-container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="header-container"
>
            <p>
                        Tony
            </p>
            <img
                        alt="user profile picture"
                        className="profile-picture"
                        src={undefined}
            />
</div>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Tony
</p>,
              <img
                alt="user profile picture"
                className="profile-picture"
                src={undefined}
/>,
            ],
            "className": "header-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tony",
              },
              "ref": null,
              "rendered": "Tony",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "user profile picture",
                "className": "profile-picture",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="problem-container"
>
            <p
                        className="section-title"
            >
                        Heres your problem
            </p>
            <p
                        className="problem-title"
            />
            <p
                        className="problem-body"
            />
            <div
                        className="category-tags"
            >
                        <button
                                    className="display-category"
                        >
                                    category
                        </button>
            </div>
</div>,
          "className": "problem",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="section-title"
>
                Heres your problem
</p>,
              <p
                className="problem-title"
/>,
              <p
                className="problem-body"
/>,
              <div
                className="category-tags"
>
                <button
                                className="display-category"
                >
                                category
                </button>
</div>,
            ],
            "className": "problem-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Heres your problem",
                "className": "section-title",
              },
              "ref": null,
              "rendered": "Heres your problem",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "problem-title",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "problem-body",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="display-category"
>
                    category
</button>,
                ],
                "className": "category-tags",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "category",
                    "className": "display-category",
                  },
                  "ref": null,
                  "rendered": "category",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="stats-container"
>
            <p
                        className="section-title"
            >
                        Heres whats happening
            </p>
            <p
                        className="stats-title"
            >
                        Nothing at this time :(
            </p>
</div>,
          "className": "stats",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="section-title"
>
                Heres whats happening
</p>,
              <p
                className="stats-title"
>
                Nothing at this time :(
</p>,
            ],
            "className": "stats-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Heres whats happening",
                "className": "section-title",
              },
              "ref": null,
              "rendered": "Heres whats happening",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Nothing at this time :(",
                "className": "stats-title",
              },
              "ref": null,
              "rendered": "Nothing at this time :(",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="header"
>
            <div
                        className="header-container"
            >
                        <p>
                                    Tony
                        </p>
                        <img
                                    alt="user profile picture"
                                    className="profile-picture"
                                    src={undefined}
                        />
            </div>
</div>,
          <div
            className="problem"
>
            <div
                        className="problem-container"
            >
                        <p
                                    className="section-title"
                        >
                                    Heres your problem
                        </p>
                        <p
                                    className="problem-title"
                        />
                        <p
                                    className="problem-body"
                        />
                        <div
                                    className="category-tags"
                        >
                                    <button
                                                className="display-category"
                                    >
                                                category
                                    </button>
                        </div>
            </div>
</div>,
          <div
            className="stats"
>
            <div
                        className="stats-container"
            >
                        <p
                                    className="section-title"
                        >
                                    Heres whats happening
                        </p>
                        <p
                                    className="stats-title"
                        >
                                    Nothing at this time :(
                        </p>
            </div>
</div>,
        ],
        "className": "frame-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="header-container"
>
              <p>
                            Tony
              </p>
              <img
                            alt="user profile picture"
                            className="profile-picture"
                            src={undefined}
              />
</div>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Tony
</p>,
                <img
                  alt="user profile picture"
                  className="profile-picture"
                  src={undefined}
/>,
              ],
              "className": "header-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tony",
                },
                "ref": null,
                "rendered": "Tony",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "user profile picture",
                  "className": "profile-picture",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="problem-container"
>
              <p
                            className="section-title"
              >
                            Heres your problem
              </p>
              <p
                            className="problem-title"
              />
              <p
                            className="problem-body"
              />
              <div
                            className="category-tags"
              >
                            <button
                                          className="display-category"
                            >
                                          category
                            </button>
              </div>
</div>,
            "className": "problem",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="section-title"
>
                  Heres your problem
</p>,
                <p
                  className="problem-title"
/>,
                <p
                  className="problem-body"
/>,
                <div
                  className="category-tags"
>
                  <button
                                    className="display-category"
                  >
                                    category
                  </button>
</div>,
              ],
              "className": "problem-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Heres your problem",
                  "className": "section-title",
                },
                "ref": null,
                "rendered": "Heres your problem",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "problem-title",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "problem-body",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="display-category"
>
                      category
</button>,
                  ],
                  "className": "category-tags",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "host",
                    "props": Object {
                      "children": "category",
                      "className": "display-category",
                    },
                    "ref": null,
                    "rendered": "category",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="stats-container"
>
              <p
                            className="section-title"
              >
                            Heres whats happening
              </p>
              <p
                            className="stats-title"
              >
                            Nothing at this time :(
              </p>
</div>,
            "className": "stats",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="section-title"
>
                  Heres whats happening
</p>,
                <p
                  className="stats-title"
>
                  Nothing at this time :(
</p>,
              ],
              "className": "stats-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Heres whats happening",
                  "className": "section-title",
                },
                "ref": null,
                "rendered": "Heres whats happening",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Nothing at this time :(",
                  "className": "stats-title",
                },
                "ref": null,
                "rendered": "Nothing at this time :(",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
